assembly/src/parsers/field_ops.rs

```diff
@@ -482,58 +512,22 @@ pub(super) fn append_pow2_op(span_ops: &mut Vec<Operation>, checked_mode: bool)
 
     span_ops.push(Operation::Push(Felt::new(2)));
     span_ops.push(Operation::Swap);
```

Comment:
Is this `swap` really needed here? It seems like we can skip it and just replace `movup.2` on line 518 with `movup.3`. Should save 1 cycle.